packages/application-shell/src/apollo-links/error-link.js

```diff
@@ -7,10 +7,31 @@ import history from '@commercetools-frontend/browser-history';
 
 // Checks response from GraphQL in order to scan 401 errors and redirect the
 // user to the login page resetting the store and showing the proper message
-const errorLink = onError(({ networkError }) => {
-  if (networkError && networkError.statusCode === STATUS_CODES.UNAUTHORIZED) {
-    history.push(`/logout?reason=${LOGOUT_REASONS.UNAUTHORIZED}`);
+const errorLink = onError(
+  ({ graphQLErrors, networkError, operation, forward }) => {
+    if (networkError && networkError.statusCode === STATUS_CODES.UNAUTHORIZED) {
+      history.push(`/logout?reason=${LOGOUT_REASONS.UNAUTHORIZED}`);
+    }
+    // In case of graphql errors, we want to retry unauthenticated requests by
+    // forcing our API to fetch a new token, using the `X-Force-Token` header.
+    // https://www.apollographql.com/docs/link/links/error/#retrying-failed-requests
+    // We need to do this as the `token-retry-link` only works for network errors.
+    // https://www.apollographql.com/docs/link/links/retry/
+    if (graphQLErrors) {
```

Comment:
Nice! Thanks for the comment above. Do you think we can slap a test at this? I remember Jan Florian back then did some work to test links and their observables.